src/mbgl/programs/symbol_program.hpp

```diff
@@ -298,6 +298,48 @@ class SymbolProgram : public SymbolProgramBase {
         return allAttributeBindings.activeCount();
     }
 
+    template <class DrawMode>
+    void draw(gfx::Context& context,
+              gfx::RenderPass& renderPass,
+              const DrawMode& drawMode,
+              const gfx::DepthMode& depthMode,
+              const gfx::StencilMode& stencilMode,
+              const gfx::ColorMode& colorMode,
+              const gfx::CullFaceMode& cullFaceMode,
+              const gfx::IndexBuffer& indexBuffer,
+              const Segment<AttributeList>& segment,
+              const UniformValues& uniformValues,
+              const AttributeBindings& allAttributeBindings,
+              const TextureBindings& textureBindings,
+              const std::string& layerID) {
+        static_assert(Primitive == gfx::PrimitiveTypeOf<DrawMode>::value, "incompatible draw mode");
```

Comment:
Extracting per segment draw to separate method seems like an orthogonal, unnecessary refactoring, maybe unintentionally in this commit? It was shorter (subjective): static_asert and "if (!program)" are already there in caller before the loop.

Reply:
thanks for clarifying this - segment's draw used from [diff](https://github.com/mapbox/mapbox-gl-native/pull/14386/files#diff-e28696fa5b68ae9576f684ddfdbaf98cR396) - failed to load it.